forge/src/Jackal.sol

```diff
@@ -49,18 +58,38 @@ abstract contract Jackal {
         return p;
     }
 
-    function postFileFrom(address from, string memory merkle, uint64 filesize) public payable{
-        require(msg.sender != address(0), "Invalid sender address");
-        require(getAllowance(msg.sender, from), "No allowance for this contract set");
-
-        uint256 pE = getStoragePrice(filesize);
+    function postFileFrom(address from, string memory merkle, uint64 filesize, string memory note, uint64 expires)
+        public
+        payable
+        validAddress
+        hasAllowance(from)
+    {
+        require(expires >= 30);
+        uint256 pE = getStoragePrice(filesize, 2400); // 12 * 200 months
+        require(msg.value >= pE, string.concat("Insufficient payment, need ", Strings.toString(pE), " wei"));
```

Comment:
We will also need to make sure that if `expires == 0` then we don't require any funds be sent over since they'll be using their storage account space instead of buying the file outright.